src/luaharfbuzz/subset-input.c

```diff
@@ -0,0 +1,148 @@
+
+
+#include <hb-subset.h>
+
+#include "luaharfbuzz.h"
+
+int subset_input_new(lua_State *L) {
+  hb_subset_input_t *input;
+  input = hb_subset_input_create_or_fail();
+  if (input == NULL) {
+    lua_pushnil(L);
+  } else {
+    hb_subset_input_t **ud =
+        (hb_subset_input_t **)lua_newuserdata(L, sizeof(hb_subset_input_t *));
+    *ud = input;
+    luaL_getmetatable(L, "harfbuzz.SubsetInput");
+    lua_setmetatable(L, -2);
+  }
+  return 1;
+}
+
+// hb_subset_input_pin_axis_location
+static int subset_input_pin_axis_location(lua_State *L) {
+  // arguments: input, face, tag, value
+  hb_subset_input_t *input =
+      *(hb_subset_input_t **)luaL_checkudata(L, 1, "harfbuzz.SubsetInput");
+  Face *face = (Face *)luaL_checkudata(L, 2, "harfbuzz.Face");
+  if (*face == NULL) {
+    luaL_error(L, "Face is NULL");
+    return 0;
+  }
+  if (hb_face_get_empty() == *face) {
+    luaL_error(L, "Face is empty");
+    return 0;
+  }
+  if (hb_face_get_glyph_count(*face) == 0) {
+    luaL_error(L, "Face has no glyphs");
+    return 0;
+  }
+  if (lua_type(L, 3) != LUA_TUSERDATA) {
+    luaL_error(L, "Expected Tag userdata");
+    return 0;
+  }
+  if (lua_type(L, 4) != LUA_TNUMBER) {
+    luaL_error(L, "Expected number for axis value");
+    return 0;
+  }
```

Comment:
I don’t think the check here are needed. `luaL_checkudata` and `luaL_checknumber` below will take care of it.